### PR TITLE
batcheval: make ExportReq tests run on virtual clusters

### LINKED ISSUE #109429: kvserver/batcheval: TestExportCmd misbehaves when pointed to a secondary tenant
**Describe the problem**

When `TestExportCmd` is pointed to a secondary tenant, it fails with the following errors:
```
=== RUN   TestExportCmd
    test_log_scope.go:167: test logs captured to: /tmp/_tmp/ec3008d890cd291dec0f79cd501ac5a1/logTestExportCmd519896531
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:129: test server using tenant; see comment at top of test_server_shim.go for details.
=== RUN   TestExportCmd/ts1
    cmd_export_test.go:225:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:225
                Error:          "[]" should have 1 item(s), but has 0
                Test:           TestExportCmd/ts1
                Messages:       unexpected files in latest export
=== RUN   TestExportCmd/ts2
=== RUN   TestExportCmd/ts3
    cmd_export_test.go:245:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:245
                Error:          "[]" should have 1 item(s), but has 0
                Test:           TestExportCmd/ts3
                Messages:       unexpected files in latest export
=== RUN   TestExportCmd/ts4
    cmd_export_test.go:252:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:252
                Error:          "[]" should have 1 item(s), but has 0
                Test:           TestExportCmd/ts4
                Messages:       unexpected files in latest export
=== RUN   TestExportCmd/ts5
    cmd_export_test.go:267:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:267
                Error:          "[]" should have 2 item(s), but has 0
                Test:           TestExportCmd/ts5
                Messages:       unexpected files in latest export
=== RUN   TestExportCmd/ts6
    cmd_export_test.go:290:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:290
                Error:          "[]" should have 2 item(s), but has 0
                Test:           TestExportCmd/ts6
                Messages:       unexpected files in latest export
=== RUN   TestExportCmd/ts7
    cmd_export_test.go:335:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:151
                                                        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_export_test.go:335
                Error:          "[]" should have 2 item(s), but has 0
                Test:           TestExportCmd/ts7
                Messages:       unexpected files in latest export
=== CONT  TestExportCmd
    cmd_export_test.go:431: -- test log scope end --
test logs left over in: /tmp/_tmp/ec3008d890cd291dec0f79cd501ac5a1/logTestExportCmd519896531

```

**Expected behavior**

The tests should be modified to use a tenant prefix in the KV key boundaries.

Epic: CRDB-26687

Jira issue: CRDB-30921

### PR DESCRIPTION
**Pull request by dt**
Release note: none.
Epic: CRDB-26687.

### CODE REVIEW

File: pkg/kv/kvserver/batcheval/cmd_export_test.go [hunk 1]
```diff
@@ -55,14 +55,26 @@ func TestExportCmd(t *testing.T) {
...
 
+	// sKey takes a string key and makes a key, adding a tenant prefix if needed.
+	sKey := func(s string) roachpb.Key {
+		if ts.Codec().ForSystemTenant() {
+			return []byte(s)
+		}
+		if s == keys.MaxKey.String() {
+			return ts.Codec().TenantSpan().EndKey
+		}
+		return []byte(ts.Codec().TenantSpan().Key.String() + s)
```
Comment (knz) [inline pkg/kv/kvserver/batcheval/cmd_export_test.go:75]: I find this more readable: `return append(ts.Codec().TenantPrefix(), s...)`

(also what i've used in other places)

File: pkg/kv/kvserver/batcheval/cmd_export_test.go [hunk 2]
```diff
@@ -350,20 +364,21 @@ INTO
 		expect(t, res7, 1, 1, 1, 1)
 		latestRespHeader = kvpb.ResponseHeader{
 			ResumeSpan: &roachpb.Span{
-				Key:    []byte(fmt.Sprintf("/Table/%d/1/2", tableID)),
-				EndKey: []byte("/Max"),
+				Key:    sKey(fmt.Sprintf("/Table/%d/1/2", tableID)),
+				EndKey: sKey("/Max"),
```
Comment (knz) [inline pkg/kv/kvserver/batcheval/cmd_export_test.go:368]: Wait how is that even supposed to work?

`sKey()` takes a string as argument _but does not interpret it_.
So this is literally encoding the bytes `'/', 'T', 'a', 'b'...` as start key, and the bytes `'/', 'M', 'a'..` as end key.

'M' sorts before 'T', so this span is really empty.

Did you mean to construct proper table prefixes and keys.KeyMax here?

Comment (dt) [reply pkg/kv/kvserver/batcheval/cmd_export_test.go:368]: > Did you mean to construct proper table prefixes and keys.KeyMax here?

I did not. 

If look at the code _before_ my change, it is using these literal strings `/Max` and `/Table`, passing them as the "expected" arguments to comparisons. My changes here are not changing that, just adding `/Tenant/x` as needed.

I started down the path of using `keys.Rewrite.*ToTenantPrefix`, and which then would have meant switching the literals to e.g. `keys.KeyMax` , but this was a more invasive change to the existing test than is required just to remove the tenant/non-tenant limit.

File: pkg/kv/kvserver/batcheval/cmd_export_test.go [hunk 3]
```diff
@@ -299,7 +311,7 @@ INTO
 		// the max overage being exceeded.
 		defer resetMaxOverage(t)
 		setMaxOverage(t, "'1b'")
-		const expectedError = `export size \(11 bytes\) exceeds max size \(2 bytes\)`
+		const expectedError = `export size \(\d\d bytes\) exceeds max size \(2 bytes\)`
```
Comment (stevendanna) [inline pkg/kv/kvserver/batcheval/cmd_export_test.go:314]: Maybe \d+ to be done with it?

**Review by stevendanna — APPROVED** [review]
As you and knz discussed, this test could probably be improved, but this still seems like a strict improvement.

**Review by knz — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110193)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @stevendanna)

___
*[`pkg/kv/kvserver/batcheval/cmd_export_test.go` line 368 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/110193#gh-1318963795:-NfotrBL1BgI877oeWHC:b5xck4a) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9541c7b7fe6f16aa6f6daf6102828a83d7343ff/pkg/kv/kvserver/batcheval/cmd_export_test.go#L368)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

> Did you mean to construct proper table prefixes and keys.KeyMax here?

I did not. 

If you look at the code _before_ my change, it is using these literal strings `/Max` and `/Table`, passing them as the "expected" arguments to comparisons. As you point out, EndKey is before Key here, because T is after M, but the test is only contracting this "Span" long enough for it to get made into a string to compare to the actual reply span's string representation. My changes here are not changing that -- how the test itself works -- but are just adding `/Tenant/x` as needed.

I did briefly start down the path of using `keys.Rewrite.*ToTenantPrefix`, and which then would have meant switching the literals to e.g. `keys.KeyMax` , but this was a more invasive change to the existing test than is required just to remove the tenant/non-tenant limit.
</blockquote></details>

i'm ok with this remaining no more broken than it was already, but at least you could leave an explanatory TODO explaining what's going on here.

**Pull request closed without merging**